bin/loc-budget.test.sh

```
@@ -494,4 +494,31 @@ git -C "$R" add -A
+  && ok "FACTORY_GATE skips before measuring, and prints no summary line" \
+  || bad "FACTORY_GATE skip" "rc=$rc out=$out"
+
+# --- gate-ness is never INFERRED from git failing (#15) ---
+# The same unmeasurable tree without the flag must still refuse. If a broken git
+# implied "in a gate", any broken git anywhere would silently disarm the budget.
+R="$(mkrepo)"
+printf 'x = 1\n' > "$R/a.py"
+git -C "$R" add -A
+rm -r "$R/.git"
+rc=0; out="$(LOC_PATHS='*.py' bash "$LB" "$R" 2>&1)" || rc=$?
+[ "$rc" != 0 ] && ok "an unmeasurable tree without the flag still refuses" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)